Sources/CardPayments/CardClient.swift

```diff
@@ -7,27 +7,87 @@ import CorePayments
 public class CardClient: NSObject {
 
     public weak var delegate: CardDelegate?
-
+    public weak var vaultDelegate: CardVaultDelegate?
+    
     private let apiClient: APIClient
     private let config: CoreConfig
     private let webAuthenticationSession: WebAuthenticationSession
     private var analyticsService: AnalyticsService?
+    private var graphQLClient: GraphQLClient?
 
     /// Initialize a CardClient to process card payment
     /// - Parameter config: The CoreConfig object
     public init(config: CoreConfig) {
         self.config = config
         self.apiClient = APIClient(coreConfig: config)
         self.webAuthenticationSession = WebAuthenticationSession()
+        self.graphQLClient = GraphQLClient(environment: config.environment)
     }
 
     /// For internal use for testing/mocking purpose
-    init(config: CoreConfig, apiClient: APIClient, webAuthenticationSession: WebAuthenticationSession) {
+    init(
+        config: CoreConfig,
+        apiClient: APIClient,
+        webAuthenticationSession: WebAuthenticationSession,
+        graphQLClient: GraphQLClient? = nil
+    ) {
         self.config = config
         self.apiClient = apiClient
         self.webAuthenticationSession = webAuthenticationSession
+        self.graphQLClient = graphQLClient
     }
-
+    
+    public func vault(vaultRequest: CardVaultRequest) {
+        Task {
+            guard graphQLClient != nil else {
```

Comment:
Do we need this guard? It doesn't look like we use `graphQLClient` in this method at all so not sure why we need to check that it's not nil.

Reply:
The `updateSetupToken` function that it calls uses it so I thought I'd check it early before anything else is done.